pkg/pillar/cmd/zedmanager/handledomainmgr.go

```diff
@@ -67,12 +67,12 @@ func MaybeAddDomainConfig(ctx *zedmanagerContext,
 	}
 	dc.DiskConfigList = make([]types.DiskConfig, numDisks)
 	i := 0
+	if len(aiConfig.StorageConfigList) < len(aiStatus.StorageStatusList) {
```

Comment:
What is this used for? When do u expect this error?
This essentially says the Storage Volume has been deleted - right? Such a thing should be handled in Zedmanager itself - right? Why would it come all the way to handleDomainMgr? 

Also - if you see this error, it would be nice to see which of the entries are now deleted. Otherwise it would be difficult to debug..

Reply:
I have not much idea about this check. I was going through the zedmanager code and I found this check inside for loop which doesn't make sense to me. So, I just extracted out of the loop. If this is an invalid check we can remove it.

Reply:
The reason to have the check inside the loop is to make sure the index, which is used on the next line, isn't out of bounds. The further away the check is from the dereference of the array the more likely it is to get violated as the code is changed by different hands.
Defensive programming I think it is called.

Reply:
We are checking that the storage config list should not be greater than the storage status list. If that's the case then the error message is also incorrect (currently it is "More StorageStatus than StorageConfig", it should be "More StorageConfig than StorageStatus") I will correct it. And can't we check the length of both storage config and the status list just before the loop?

Reply:
> The reason to have the check inside the loop is to make sure the index, which is used on the next line, isn't out of bounds. The further away the check is from the dereference of the array the more likely it is to get violated as the code is changed by different hands.
> Defensive programming I think it is called.

I am not sure why this was changed in the first place.. I am ok with the way he changed it too - Range cannot be greater than Status list after his check. 

@zed-rishabh - Is this more of a random cleanup or did you see some issue with the code?

Reply:
@kalyan-nidumolu It is a random clean up and I didn't face any issues.